### DISCUSSION TITLE
How to get selected rows from the table, outside of table?

### DISCUSSION THREAD

**Discussion by AndreSaSilva** [filamentphp/filament#14206] [Help]
### Package

Panel builder

### Package Version

v3

### How can we help you?

I have a custom page with a form and a table. The data in the table is updated based on the selection of the Industry field in the form.
My goal is to select multiple rows in the table and, when saving the form, create a record for each row selected in the table.
My problem is figuring out how to get the selected rows from the table.
Can anyone help me or give me an idea of how to solve this problem?

![image](https://github.com/user-attachments/assets/6a9445da-8e44-4895-8397-3de1acddea4e)

**Comment by ModestasV** [comment]
Hey,

It's pretty hard to exactly understand the issue you are having, but based on:

- Selecting multiple records
- Creating new records under the selected ones

I think that bulk action would be a better thing to do here (instead of a separate form page) 

https://filamentphp.com/docs/3.x/tables/actions#bulk-actions

In this case, you could simple create an action with a form and on submit - iterate over each record:

```
BulkAction::make('delete')
    ->requiresConfirmation()
    ->action(fn (Collection $records) => $records->each->delete())
```

Of course, this is for delete (from documentation), but this can be done like so:

```
Tables\Actions\BulkAction::make('Update')
                    ->form([
                        Forms\Components\TextInput::make('invoice_number_prefix')
                    ])
                    ->action(function ($records, $data) {
                        foreach ($records as $record) {
                            $record->invoice_number = $data['invoice_number_prefix'] . $record->invoice_number;
                            $record->save();
                        }
                    })
```

Hope that helps!

**Comment by elmudometal** [comment]
The way is through a bulk action, and here you can see how I solve it: [https://github.com/elmudometal/filament-select-table](url)
